### PR TITLE
Inliner: updates to RandomPolicy

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Fix issues with RandomPolicy setup in release builds with -DINLINE_DATA.

Reparent RandomPolicy on top of DiscretionaryPolicy to enable inline
data dumps from random inline runs. Remove some now-unneeded overrides.

Add a full dump mode to JitInlineDumpData that reports the inliner-visible
data for all inlines, as opposed to inliner-visible and post-inline
data for just the most recent inline. Small mods to the dumper code to
adjust comma placement for this new mode.

Have the RandomPolicy make the full set of profitability observations
for each accepted inline so they can be dumped.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@dotnet/jit-contrib PTAL

**Comment by AndyAyersMS** [comment]
Looks like the [Windows Pri1 tests](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_release_windows_nt_pri1_prtest/95/) hit some kind of build race. Will retry...
```
error MSB3501: Could not read lines from file "D:\j\workspace\x64_release_w---0575cb46\bin\int\Managed\GC\Stress\Tests\MulDimJagAry\MulDimJagAry.csproj.FileListAbsolute.txt". The process cannot access the file 'D:\j\workspace\x64_release_w---0575cb46\bin\int\Managed\GC\Stress\Tests\MulDimJagAry\MulDimJagAry.csproj.FileListAbsolute.txt' because it is being used by another process. [D:\j\workspace\x64_release_w---0575cb46\tests\src\GC\Stress\Tests\MulDimJagAry.csproj]
```

@dotnet-bot retest Windows_NT x64 Release Priority 1 Build and Test

**Comment by AndyAyersMS** [comment]
@JosephTremoulet PTAL

File: src/jit/inline.cpp [hunk 1]
```diff
@@ -500,14 +500,25 @@ void InlineContext::DumpXml(FILE* file, unsigned indent)
         fprintf(file, "%*s<Offset>%u</Offset>\n", indent + 2, "", offset);
         fprintf(file, "%*s<Reason>%s</Reason>\n", indent + 2, "", inlineReason);
 
-        // Optionally, dump data about the last inline
-        if ((JitConfig.JitInlineDumpData() != 0) && (this == m_InlineStrategy->GetLastContext()))
+        // Optionally, dump data about the inline
+        const int dumpDataSetting = JitConfig.JitInlineDumpData();
+
+        // JitInlineDumpData=1 -- dump data plus deltas for last inline only
```
Comment (JosephTremoulet) [inline src/jit/inline.cpp]: Thanks for the descriptions of what "1" and "2" mean here.

File: src/jit/inline.cpp [hunk 2]
```diff
@@ -1456,10 +1467,20 @@ void InlineStrategy::DumpXml(FILE* file, unsigned indent)
...
         fprintf(file, "<Policy>%s</Policy>\n", m_LastSuccessfulPolicy->GetName());
 
-        if (JitConfig.JitInlineDumpData() != 0)
+        const int dumpDataSetting = JitConfig.JitInlineDumpData();
+        if (dumpDataSetting != 0)
         {
             fprintf(file, "<DataSchema>");
-            DumpDataSchema(file);
+
+            if (dumpDataSetting == 1)
```
Comment (JosephTremoulet) [inline src/jit/inline.cpp]: I think that minimally this code should also have the descriptions of what "1" and "2" mean, or preferably that you should go ahead and define an enum.

**Comment by AndyAyersMS** [comment]
Latest test failures ([here](https://ci.dot.net/job/dotnet_coreclr/job/master/job/checked_osx_flow_prtest/154/) and [here](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x86_ryujit_checked_windows_nt_prtest/138/)) appear to be infrastructure issues (machine went offline). Will retry.

@dotnet-bot retest OSX x64 Checked Build and Test
@dotnet-bot retest Windows_NT x86 ryujit Checked Build and Test

**Pull request merged**